libraries/AP_NavEKF2/AP_NavEKF2.cpp

```diff
@@ -1358,9 +1358,16 @@ void  NavEKF2::getFilterGpsStatus(int8_t instance, nav_gps_status &status) const
 // send an EKF_STATUS_REPORT message to GCS
 void NavEKF2::send_status_report(mavlink_channel_t chan) const
 {
-    if (core) {
-        core[primary].send_status_report(chan);
+    if (!core) {
+        return;
+    }
+    for (uint8_t i=0; i<num_cores; i++) {
+        if (i == primary) {
+            continue;
+        }
+        core[i].send_status_report(chan);
```

Comment:
Ahhh, please no! This dies so badly with full links. This is violating the single message per interval rule again on us. What this ends up doing is on a full link. Even worse though you missed that your only space checking is up in the GCS_MAVLink class.... All your later messages will corrupt the MAVLink stream, and lead to packet loss.